src/cargo/util/toml.rs

```diff
@@ -737,7 +737,12 @@ impl TomlManifest {
                -> CargoResult<Vec<(PackageIdSpec, Dependency)>> {
         let mut replace = Vec::new();
         for (spec, replacement) in self.replace.iter().flat_map(|x| x) {
-            let spec = try!(PackageIdSpec::parse(spec));
+            let spec = match PackageIdSpec::parse(spec) {
+                Ok(spec) => spec,
+                Err(_)   => bail!("replacements must specify a \
+                                   valid semver version \
+                                   to replace, but `{}` does not", spec),
+            };
```

Comment:
@alexcrichton yes, that sounds like a good practice. I made the change here https://github.com/rust-lang/cargo/pull/3139/commits/89b2d267244b244776aa12bbae9bd0441a350ef6